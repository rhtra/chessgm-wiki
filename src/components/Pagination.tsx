type Props = {
  page: number;
  totalPages: number;
  onChange: (page: number) => void;
};

export default function Pagination({ page, totalPages, onChange }: Props) {
  if (totalPages <= 1) return null;

  return (
    <div className="pagination">
      <button
        disabled={page === 1}
        onClick={() => onChange(page - 1)}
        className="btn"
      >
        ← Prev
      </button>

      {Array.from({ length: totalPages }, (_, i) => (
        <button
          key={i + 1}
          className={`page-btn ${page === i + 1 ? "active" : ""}`}
          onClick={() => onChange(i + 1)}
        >
          {i + 1}
        </button>
      ))}

      <button
        disabled={page === totalPages}
        onClick={() => onChange(page + 1)}
        className="btn"
      >
        Next →
      </button>
    </div>
  );
}
