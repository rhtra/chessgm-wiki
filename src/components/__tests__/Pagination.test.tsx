import { render } from "@testing-library/react";
import Pagination from "../Pagination";

describe("Pagination snapshots", () => {
  it("matches snapshot on first page", () => {
    const { asFragment } = render(
      <Pagination page={1} totalPages={3} onChange={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot on middle page", () => {
    const { asFragment } = render(
      <Pagination page={2} totalPages={5} onChange={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
