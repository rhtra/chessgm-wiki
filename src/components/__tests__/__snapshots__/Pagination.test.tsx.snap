// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pagination snapshots matches snapshot on first page 1`] = `
<DocumentFragment>
  <div
    class="pagination"
  >
    <button
      class="btn"
      disabled=""
    >
      ← Prev
    </button>
    <button
      class="page-btn active"
    >
      1
    </button>
    <button
      class="page-btn "
    >
      2
    </button>
    <button
      class="page-btn "
    >
      3
    </button>
    <button
      class="btn"
    >
      Next →
    </button>
  </div>
</DocumentFragment>
`;

exports[`Pagination snapshots matches snapshot on middle page 1`] = `
<DocumentFragment>
  <div
    class="pagination"
  >
    <button
      class="btn"
    >
      ← Prev
    </button>
    <button
      class="page-btn "
    >
      1
    </button>
    <button
      class="page-btn active"
    >
      2
    </button>
    <button
      class="page-btn "
    >
      3
    </button>
    <button
      class="page-btn "
    >
      4
    </button>
    <button
      class="page-btn "
    >
      5
    </button>
    <button
      class="btn"
    >
      Next →
    </button>
  </div>
</DocumentFragment>
`;
